docs/adr/passthrough-rpc.md

```diff
@@ -0,0 +1,105 @@
+
+```
+{ 
+  "passthrough_rpcs": [ 
+    { 
+      "method": "device.model", 
+      "url": "ws://127.0.0.1:9992/jsonrpc", 
+      "protocol": "jsonrpc", 
+      "include_context": ["appId"] 
+    } 
+  ] 
+} 
+```
+
+At the gateway level (before any Ripple handler code), Ripple will check if the called method is a passthrough rpc. If it is then it just sends the exact jsonrpc using the given protocol through the given url. Ripple will first only support jsonrpc as a persistent websocket connection with pass-through jsonrpc. 
+Possible support in future for comrpc, and would need to come up with a common algorithm for mapping jsonrpc to comrpc.
+
+### Context passing
+
+Only Ripple knows the context of calling application. We may want the plugin that implements the jsonrpc call to need to know who the calling app is. 
+
+The array "include_context" tells Ripple it needs to include some context when passing it to the implementing component. 
+
+If an app makes the call: 
+```
+{ 
+  "jsonrpc": "2.0", 
+  "id": 1, 
+  "method": "device.model" 
+} 
+```
+
+Then if include_context contains "appId", Ripple will send this message through the jsonrpc websocket: 
+
+```
+{ 
+  "jsonrpc": "2.0", 
+  "id": 1, 
+  "method": "device.model",
+  "params": {
+    "__ctx": {
+      "appId": "test.firecert"
+    }
+  }
+} 
+```
+
+`__ctx` is a reserved parameter name that all passthrough providers must handle. `__ctx` will be left undefined if `include_context` is undefined or is an empty list.
+
+Only `appId` will be supported for first implementation, but the configuration specification can support other ctx keys in the future.
+
+ ### Pass-through RPC as override
+
+ Ripple must be certified for all firebolt APIs that are in the firebolt specification. So Ripple should not be completely dependent on configured pass-throughs. Ripple thus can still be the provider of the rpc handler for the APIs. `passthrough_rpcs` would be an override of those existing handlers.
+
+ If there is no registered passthrough rpc and no Ripple handler defined, then Ripple should return an error back to the app with a not supported error.  
+
+Ripple should fail to start if a MUST capability in Firebolt does not have a corresponding passthrough or handler.
+
+### Permission and grant checking
+
+The passthrough provider does not need to worry about permission checking, Ripple will continue to do that. Any request from Ripple should be considered already checked as far as permissions.
+
+## Open questions
+
+### Correlation of Requests and Response from the passthrough provider
+
+Since the jsonrpc id is also passed-through, how does Ripple know which id space the response is from?
+For example, if app1 makes a "methodA" call with jsonrpc id=1 and app2 makes a "methodB" call also with jsonrpc id=1 and both are sent through the passthrough. When the response comes back with id=1, how does Ripple know the response is meant for app1 or app2.
```

Comment:
Can we preload a correlationId similar to the provider broker into the `__ctx`. Correlation Id can map back to the connection id for the rpc router to send that back depending on the transport